### PR TITLE
Pass -UserScope as required by RunUpdateHelpTests

### PR DESCRIPTION
**Pull request by yecril71pl**
# PR Summary

The function `RunUpdateHelpTests` does not expect to get ~**Scope**~ but **UserScope**.  I modified the calls accordingly.

## PR Context

I am lost why passing a **Scope** to a function that accepts **UserScope** would have any good results.  I think the current behaviour is that the tests are skipped because `$userscope -eq $false`, as evidenced by the following [test log entry](https://dev.azure.com/powershell/PowerShell/_build/results?buildId=58204&view=ms.vss-test-web.build-test-results-tab): 

> Validate Update-Help -SourcePath for all PowerShell modules for user scope..Validate Update-Help for module 'Microsoft.PowerShell.Core' with scope as 'False'

… and erroneously marked as passed.

## PR Checklist

- [X] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [X] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [X] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [X] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [X] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [X] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 [hunk 1]
```diff
@@ -199,7 +199,7 @@ function RunUpdateHelpTests
         if ($powershellCoreModules -contains $moduleName)
         {
 
-            It "Validate Update-Help for module '$moduleName' with scope as '$userscope'" -Skip:(!(Test-CanWriteToPsHome) -and $userscope -eq $false) {
+            It "Validate Update-Help for module '$moduleName' with user scope as '$userscope'" -Skip:(!(Test-CanWriteToPsHome) -and $userscope -eq $false) {
```
Comment (iSazonov) [inline test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:202]: It is not correct. Please try something like:
```suggestion
            It "Validate Update-Help for module '$moduleName' with $(if ($userscope) 'User' else 'AllScope') scope" -Skip:(!(Test-CanWriteToPsHome) -and $userscope -eq $false) {
```

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 [hunk 2]
```diff
@@ -359,7 +359,7 @@ Describe "Validate Update-Help from the Web for one PowerShell module for user s
         $ProgressPreference = $SavedProgressPreference
     }
 
-    RunUpdateHelpTests -Tag "CI" -Scope 'CurrentUser'
+    RunUpdateHelpTests -Tag:CI -UserScope
```
Comment (iSazonov) [inline test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:362]: Please use the common pattern here and below:
```suggestion
    RunUpdateHelpTests -Tag "CI" -UserScope
```

**Review by iSazonov — COMMENTED** [review]
We lost this tracking in https://github.com/PowerShell/PowerShell/pull/6352#discussion_r216417311

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 (hunk 2 above)
Comment (yecril71pl) [reply test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1]: As requested, although I have never seen a common pattern of using spurious double quotes in scripts.

Comment (iSazonov) [reply test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1]: The "CI" string is often used in our tests, It is more easy discover one pattern if we will need to change this.

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 [hunk 3]
```diff
@@ -199,18 +199,18 @@ function RunUpdateHelpTests
...
-                {
-                    $params = @{Path = $testCases[$moduleName].HelpInstallationPathHome}
-                    $updateScope = @{Scope = 'CurrentUser'}
-                }
-                else
-                {
-                    $params = @{Path = $testCases[$moduleName].HelpInstallationPath}
-                    $updateScope = @{Scope = 'AllUsers'}
-                }
+            It "Validate Update-Help for module '$moduleName' in '$updateScope'" -Skip:(!(Test-CanWriteToPsHome) -and $userscope -eq $false) {
```
Comment (iSazonov) [inline test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:213]: '$updateScope' returns "System.Collections.Hashtable". Yes?

Comment (yecril71pl) [reply test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:213]: I am deeply ashamed of myself.  I assumed it would be nicely expanded in a string.  I am working on it.

Comment (iSazonov) [reply test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:213]: We could simply add new variable for the test name.

Comment (yecril71pl) [reply test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1]: I would rather be as deep to the metal as possible in a test log.

**Comment by adityapatwardhan** [comment]
@yecril71pl Thank you for your PR. Please have a look at the failures.

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 [hunk 4]
```diff
@@ -5,7 +5,7 @@ Import-Module HelpersCommon
 
 # Test Settings:
 # This is the list of PowerShell modules for which we test update-help
-$powershellCoreModules = @(
+[string[]] $powershellCoreModules = @(
```
Comment (iSazonov) [inline test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:8]: Please revert the unrelated change. Below too.

I understand that you have your own scripting style, however a general practice is to follow the style of the file and the traditions of the project.
Also please do not mix functional and styling changes and unrelated changes in one PR.

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 [hunk 5]
```diff
@@ -26,119 +26,127 @@ $powershellCoreModules = @(
...
 {
     $extension = ".cab"
 }
 
+[string] $userHelpRoot = $null
+
 if([System.Management.Automation.Platform]::IsWindows)
 {
-    $userHelpRoot = Join-Path $HOME "Documents/PowerShell/Help/"
+    # To the reader: the explicit parameter names below are required by a brainless code checker.
```
Comment (iSazonov) [inline test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:40]: Please remove the unneeded comment.

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 [hunk 6]
```diff
@@ -26,119 +26,127 @@ $powershellCoreModules = @(
...
+    $userHelpRoot = Join-Path -Path:$HOME -ChildPath:Documents -AdditionalChildPath:PowerShell, Help
```
Comment (iSazonov) [inline test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:41]: Please follow the file style. Below too.
```suggestion
    $userHelpRoot = Join-Path -Path $HOME -ChildPath:Documents -AdditionalChildPath PowerShell, Help
```

**Review by iSazonov — DISMISSED** [review]
@yecril71pl 99% of the PR is unrelated changes. Please fix only UserScope issue and move rest of changes in follow PR.

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 (hunk 4 above)
Comment (yecril71pl) [reply test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:8]: If I fix only **UserScope**, the tests will fail because they will actually run.  The only reason they have not failed yet is that they were effectively disabled.

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 (hunk 5 above)
Comment (yecril71pl) [reply test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:40]: The comment is here so that a future editor will know that they cannot remove the parameter names, which is what anyone would do seeing this absurd syntax.

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 (hunk 4 above)
Comment (andschwa) [reply test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:8]: I agree with @yecril71pl, this is fine and looks like it was necessary.

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 (hunk 5 above)
Comment (andschwa) [reply test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:40]: Makes sense to me, thanks for the note.

File: test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1 [hunk 7]
```diff
@@ -431,7 +400,7 @@ Describe "Validate Update-Help -SourcePath for all PowerShell modules for user s
         $ProgressPreference = $SavedProgressPreference
     }
 
-    RunUpdateHelpTests -Tag "Feature" -useSourcePath -Scope 'CurrentUser'
+    RunSaveHelpTests -Tag "Feature" -useSourcePath -UserScope
```
Comment (andschwa) [inline test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1:403]: I don't mind the "style" changes in the rest of the PR, they look like they helped with debugging.

However, here's an actual bug. This test is supposed to validate `Update-Help` (which takes a `-UserScope` switch) but was erroneously changed to `RunSaveHelpTests` (which is covered by two tests below).
```suggestion
    RunUpdateHelpTests -Tag "Feature" -useSourcePath -UserScope
```

**Review by andyleejordan — CHANGES_REQUESTED** [review]
I resolved the other requested changes as I don't think they're necessary and looked helpful for getting the tests to run again. However, spotted one real bug at the end that needs fixing (the test is no longer validating the correct function).

**Comment by yecril71pl** [comment]
Wow, long time no see :smile: Is there anything you want me to do about this?  I sincerely hope it will move forward without me :wink:

**Review by andyleejordan — APPROVED** [review]
It's all good @yecril71pl! CI passed with my bug-fix, approving. Thanks for your contribution!
